cpp/tests/io/json_writer.cpp

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_wrapper.hpp>
+#include <cudf_test/iterator_utilities.hpp>
+
+#include <cudf/io/json.hpp>
+#include <cudf/io/types.hpp>
+#include <cudf/types.hpp>
+
+#include <rmm/device_uvector.hpp>
+
+#include <string>
+#include <vector>
+
+struct JsonWriterTest : public cudf::test::BaseFixture {
+};
+
```

Comment:
Please include tests with empty table and a table with empty columns; ignore if this is already covered and I missed it.

Reply:
added empty test cases and more.